spark_expectations/utils/validate_rules.py

```diff
@@ -0,0 +1,166 @@
+                f"rule: '{rule.get('rule')}' | expectation: '{expectation}' → {e}"
+            )
+
+    @staticmethod
+    def validate_agg_dq_expectation(df: DataFrame, rule: Dict) -> None:
+        """
+        Validates an agg_dq expectation by ensuring it includes aggregate functions.
+
+        Args:
+            df (DataFrame): The input DataFrame.
+            rule (Dict): Rule containing the 'expectation' and 'rule'.
+
+        Raises:
+            SparkExpectationsInvalidAggDQExpectationException: If no aggregate function is found or expression fails.
+        """
+        expectation = rule.get("expectation").lower()
+        allowed_funcs = ["sum", "avg", "min", "max", "stddev", "count"]
+
+        # Add this check to catch query-like expectations
+        if re.search(r"\bselect\b.*\bfrom\b", expectation, re.IGNORECASE):
+            raise SparkExpectationsInvalidAggDQExpectationException(
+                f"[agg_dq] Rule '{rule.get('rule')}' contains a SQL query (not allowed in agg_dq): {expectation}"
+            )
+
+        if not any(func in expectation for func in allowed_funcs):
+            raise SparkExpectationsInvalidAggDQExpectationException(
+                f"[agg_dq] Rule '{rule.get('rule')}' does not contain a valid aggregate function: {expectation}"
+            )
+        try:
+            df.selectExpr(expectation).limit(1)
+        except Exception as e:
+            raise SparkExpectationsInvalidAggDQExpectationException(
+                f"[agg_dq] Rule failed validation | rule_type: agg_dq | rule: '{rule.get('rule')}' | "
+                f"expectation: '{expectation}' → {e}"
+            )
+
+    @staticmethod
+    def validate_query_dq_expectation(df: DataFrame, rule: Dict, spark: SparkSession) -> None:
+        """
+        Validates a query_dq expectation by registering temp views and running scalar SQL.
+
+        Args:
+            df (DataFrame): The input DataFrame.
+            rule (Dict): Rule containing the 'expectation' SQL.
+            spark (SparkSession): Spark session.
+
+        Raises:
+            SparkExpectationsInvalidQueryDQExpectationException: If the SQL query is invalid or fails to run.
+        """
+        query = rule.get("expectation").lower()
+        temp_view_name = f"__dq_validation_temp_{uuid.uuid4().hex}__"
+        # Replace {placeholder} table names
+        query = re.sub(r"\{[^}]+\}", temp_view_name, query)
+
+        # Extract table names
+        table_names = SparkExpectationsValidateRules.extract_table_names_from_sql(query.lower())
+
+        # Replace all extracted table names in the query with the temp view name
+        for table_name in table_names:
+            query = re.sub(rf"\b{re.escape(table_name)}\b", temp_view_name, query)
+
+        try:
+            # Create a temporary view for validation
+            df.createOrReplaceTempView(temp_view_name)
+            # Validate the query
+            spark.sql(f"SELECT {query} AS result").collect()
```

Comment:
is it possible to capture exception on existing `expectation` execution flow or we had to create new TempView for every rule ? I feel like it can end up that we put a lot of extra burden on DBX driver

Reply:
@ManasaKanathur 🤔 maybe we can use `EXPLAIN query`  to check validity without actually running expected queries (https://spark.apache.org/docs/latest/sql-ref-syntax-qry-explain.html) 

@asingamaneni 👀 what is your take on this PR